src/io/flutter/inspector/InspectorService.java

```diff
@@ -36,8 +57,127 @@
  * inspector code running in the IDE.
  */
 public class InspectorService implements Disposable {
+
+  public static class Location {
+
+    public Location(VirtualFile file, int line, int column, int offset) {
+      this.file = file;
+      this.line = line;
+      this.column = column;
+      this.offset = offset;
+    }
+
+    private final VirtualFile file;
+    public final int line;
+    public final int column;
+    private final int offset;
+
+    public int getLine() {
+      return line;
+    }
+
+    public int getColumn() {
+      return column;
+    }
+
+    public int getOffset() {
+      return offset;
+    }
+
+    public VirtualFile getFile() {
+      return file;
+    }
+
+    public String getPath() {
+      return toSourceLocationUri(file.getPath());
```

Comment:
do we need to guard against a `null` file here?

Reply:
We don't. Added some missing @NotNull annotations to clarify.